llvm/tools/llvm-mca/Views/InstructionInfoView.cpp

```diff
@@ -29,82 +61,152 @@ void InstructionInfoView::printView(raw_ostream &OS) const {
-    if (IIVDEntry.Latency < 10)
-      TempStream << "  ";
-    else if (IIVDEntry.Latency < 100)
-      TempStream << ' ';
-
+    FOS.PadToColumn(paddings[0] + 1);
+    FOS << IIVDEntry.NumMicroOpcodes;
+    FOS.PadToColumn(paddings[1] + 1);
+    FOS << IIVDEntry.Latency;
+    FOS.PadToColumn(paddings[2]);
     if (IIVDEntry.RThroughput) {
       double RT = *IIVDEntry.RThroughput;
-      TempStream << format("%.2f", RT) << ' ';
-      if (RT < 10.0)
-        TempStream << "  ";
-      else if (RT < 100.0)
-        TempStream << ' ';
+      FOS << format("%.2f", RT);
     } else {
-      TempStream << " -     ";
+      FOS << " -";
+    }
+    FOS.PadToColumn(paddings[3] + 1);
+    FOS << (IIVDEntry.mayLoad ? "*" : " ");
+    FOS.PadToColumn(paddings[4] + 1);
+    FOS << (IIVDEntry.mayStore ? "*" : " ");
+    FOS.PadToColumn(paddings[5] + 1);
+    FOS << (IIVDEntry.hasUnmodeledSideEffects ? "U" : " ");
+    unsigned LastPaddingIdx = 5;
+
+    if (PrintFullInfo) {
+      FOS.PadToColumn(paddings[LastPaddingIdx + 1] + 1);
+      FOS << IIVDEntry.Bypass;
+      FOS.PadToColumn(paddings[LastPaddingIdx + 2] + 1);
+      FOS << IIVDEntry.Resources;
+      FOS.PadToColumn(paddings[LastPaddingIdx + 3] + 1);
+      FOS << IIVDEntry.OpcodeName;
+      LastPaddingIdx += 3;
     }
-    TempStream << (IIVDEntry.mayLoad ? " *     " : "       ");
-    TempStream << (IIVDEntry.mayStore ? " *     " : "       ");
-    TempStream << (IIVDEntry.hasUnmodeledSideEffects ? " U     " : "       ");
 
     if (PrintBarriers) {
-      TempStream << (LoweredInsts[Index]->isALoadBarrier() ? " *     "
-                                                           : "       ");
-      TempStream << (LoweredInsts[Index]->isAStoreBarrier() ? " *     "
-                                                            : "       ");
+      FOS.PadToColumn(paddings[LastPaddingIdx + 1] + 1);
+      FOS << (LoweredInsts[Index]->isALoadBarrier() ? "*" : " ");
+      FOS.PadToColumn(paddings[LastPaddingIdx + 2] + 1);
+      FOS << (LoweredInsts[Index]->isAStoreBarrier() ? "*" : " ");
+      LastPaddingIdx += 2;
     }
 
     if (PrintEncodings) {
       StringRef Encoding(CE.getEncoding(Index));
       unsigned EncodingSize = Encoding.size();
-      TempStream << " " << EncodingSize
-                 << (EncodingSize < 10 ? "     " : "    ");
-      TempStream.flush();
-      formatted_raw_ostream FOS(TempStream);
+      FOS.PadToColumn(paddings[LastPaddingIdx + 1] + 1);
+      FOS << EncodingSize;
+      FOS.PadToColumn(paddings[LastPaddingIdx + 2]);
       for (unsigned i = 0, e = Encoding.size(); i != e; ++i)
         FOS << format("%02x ", (uint8_t)Encoding[i]);
-      FOS.PadToColumn(30);
-      FOS.flush();
+      LastPaddingIdx += 2;
     }
-
-    TempStream << printInstructionString(Inst) << '\n';
+    FOS.PadToColumn(paddings[LastPaddingIdx + 1]);
+    FOS << printInstructionString(Inst);
+    if (PrintFullInfo) {
+      getComment(Inst, CommentString);
```

Comment:
nit: can we move the declaration of `CommentString` to here, I think these two lines are the only users.